### PR TITLE
Add a more structured AttestationResult.

### PR DESCRIPTION
**Pull request by setrofim**
Add a more structured implementation of AttestationResults.

Note: at the moment not all fields are properly populated. This requires
a more intrusive reimplementation and will be done in a future commit.

### CODE REVIEW

File: common/label.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	if err == nil {
+		l.strValue = strVal
+		l.isInt = false
+	}
+
+	return fmt.Errorf("could not extract Label from CBOR: %v", data)
+}
+
+// IntLabel creates a new Label from the specified int value.
+func IntLabel(value int) Label {
```
Comment (thomas-fossati) [inline common/label.go:116]: (just a note) the olden golang way is to prefix the constructor with `New`

File: common/result.go [hunk 2]
```diff
@@ -3,18 +3,133 @@
...
+	ConfigIntegrity      Status `cbor:"3,keyasint" json:"config-integrity" binding:"required"`
+	RuntimeIntegrity     Status `cbor:"4,keyasint" json:"runtime-integrity" binding:"required"`
+	CertificationStatus  Status `cbor:"5,keyasint" json:"certification-status" binding:"required"`
+}
+
+// Space for $$ar-trust-vector-extension
+type TrustVectorExtension struct {
+}
+
+// ClaimsMap contains claims extracted or derived form evidence
```
Comment (thomas-fossati) [inline common/result.go:100]:
```suggestion
// ClaimsMap contains claims extracted or derived from evidence
```

File: common/result_test.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		},
+		RawEvidence:       []byte{0xDE, 0xAD, 0xBE, 0xEF},
+		Timestamp:         time.Date(1970, 1, 1, 0, 0, 0, 0, time.UTC),
+		EndorsedClaims:    EndorsedClaims{},
+		AppraisalPolicyID: "test-policy",
+	}
+
+	bytes, err := result.ToJSON()
+	require.Nil(err)
+	assert.Equal(`{"veraison-processed-evidence":{},"status":1,"trust-vector":{"hw-authenticity":1,"sw-integrity":1,"sw-up-to-dateness":1,"config-integrity":1,"runtime-integrity":1,"certification-status":2},"raw-evidence":"3q2+7w==","timestamp":"1970-01-01T00:00:00Z","endorsed-claims":{"hw-details":{},"sw-details":{},"certification-details":{},"config-details":{}},"appraisal-policy-id":"test-policy"}`, string(bytes))
```
Comment (thomas-fossati) [inline common/result_test.go:56]: (note) you could use [JSONEq](https://pkg.go.dev/github.com/stretchr/testify/assert#JSONEq)

**Review by thomas-fossati — APPROVED** [review]
LGTM!

One small thing: have you run a `make check` in the `docs/api/challenge-response` folder?  I am asking because it's not run automatically -- we should add the rule and tooling to the CI...

**Comment by setrofim** [comment]
> One small thing: have you run a `make check` in the `docs/api/challenge-response` folder? I am asking because it's not run automatically -- we should add the rule and tooling to the CI...

Whoops,  nope, haven't done that. Will do so, and apply any necessary fixes.

**Pull request merged**